totalRP3/Modules/Register/Companions/RegisterCompanionsProfiles.lua

```diff
@@ -572,30 +581,36 @@ TRP3_API.RegisterCallback(TRP3_Addon, TRP3_Addon.Events.WORKFLOW_ON_LOADED, func
 			onClick = companionProfileSelectionList,
 			alertIcon = "Interface\\GossipFrame\\AvailableQuestIcon",
 			adapter = function(buttonStructure, unitID)
+				-- Initialize the buttonStructure parts.
+				buttonStructure.alert = false;
+				buttonStructure.icon = TRP3_InterfaceIcons.TargetOpenCompanion;
+				buttonStructure.tooltip = loc.REG_COMPANION;
+				buttonStructure.tooltipSub = loc.REG_COMPANION_TF_NO;
+
 				local ownerID, companionID = companionIDToInfo(unitID);
 				local profile = getCompanionInfo(ownerID, companionID, unitID);
-				buttonStructure.alert = nil;
-				buttonStructure.tooltip = loc.TF_OPEN_COMPANION;
-				buttonStructure.tooltipSub = nil;
-				if ownerID == Globals.player_id then
-					if profile then
-						buttonStructure.tooltip = loc.PR_PROFILE .. ": |cff00ff00" .. profile.profileName;
-						if profile.data and profile.data.IC then
-							buttonStructure.icon = profile.data.IC;
-						end
+
+				-- Check if the pet has a profile first.
+				if profile then
+					-- Retrieve profile name.
+					local name = profile.data.NA;
+
+					-- Handle if player is the owner of this pet.
+					if ownerID == Globals.player_id then
+						buttonStructure.tooltipSub = name .. "\n\n" .. TRP3_API.FormatShortcutWithInstruction("LCLICK", loc.TF_OPEN_COMPANION) .. "\n" .. TRP3_API.FormatShortcutWithInstruction("RCLICK", loc.TF_MORE_OPTIONS);
 					else
-						buttonStructure.icon = TRP3_InterfaceIcons.Gears;
-						buttonStructure.tooltip = loc.REG_COMPANION_TF_NO;
+						buttonStructure.tooltipSub = name .. "\n\n" .. TRP3_API.FormatShortcutWithInstruction("CLICK", loc.TF_OPEN_COMPANION);
+
+						-- If pet data is unread, add alert.
+						if profile.data and not profile.data.read then
```

Comment:
```suggestion
						if not profile.data.read then
```
and as we moved that test above, no need to have it here anymore